ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/CommandTopicBackupImpl.java

```diff
@@ -132,6 +132,17 @@ public void writeRecord(final ConsumerRecord<byte[], byte[]> record) {
           "Failed to write record due to out of sync command topic and backup file: " + record);
     }
 
+    if (record == null || record.key() == null || record.value() == null) {
+      String recordString = null;
+      if (record != null) {
+        recordString = String.format("key=%s, value=%s", record.key(), record.value());
+      }
+
+      LOG.warn("Cannot write a null record or null key/value to the backup file. This could be "
```

Comment:
Ditto here: could this ever happen, if we already check for null record/key/value when writing to cmd topic?

Reply:
I removed the redundant null check from the BackupReplayFile, which is called by the CommandTopicBackupImpl. I just left the one from this file.